js/src/forum/states/NotificationListState.js

```diff
@@ -0,0 +1,104 @@
+export default class NotificationListState {
+  constructor(app) {
+    this.app = app;
+
+    this.notificationPages = [];
+
+    /**
+     * Whether or not the notifications are loading.
+     *
+     * @type {Boolean}
+     */
+    this.loading = false;
+
+    /**
+     * Whether or not there are more results that can be loaded.
+     *
+     * @type {Boolean}
+     */
+    this.moreResults = false;
+  }
+
+  getNotificationPages() {
+    return this.notificationPages;
+  }
+
+  isLoading() {
+    return this.loading;
+  }
```

Comment:
Same discussion as [brought up by @datitisev here](https://github.com/flarum/core/pull/2150#discussion_r429545622)... do we want those methods, or advertise direct access to the properties?

Methods are slightly better in terms of encapsulation (reading the values could be restricted to the state class itself), so I'd favor that?

The docblocks seem to suggest the properties are still part of the official API? Is that because of backwards compatibility?

Reply:
Docblocks are a result of me copy-pasting stuff. But yeah, I'm in favor of encapsulating here since we're not planning on always keeping the data in these state classes.